src/index.tsx

```diff
@@ -53,23 +55,34 @@ export class ImageCache {
     }
 
     on(source: CachedImageURISource, handler: CacheHandler, immutable?: boolean) {
-        const {uri} = source;
-        if (!this.cache[uri]) {
-            this.cache[uri] = {
-                source,
+        console.log('ImageCache Operations on...')
+        const {dbPath, dbProvider} = source;
+        if (!this.cache[dbPath]) {
+          console.log('Entry not in application cache');
+          dbProvider.getInstance().firebase.storage()
```

Comment:
Will be better to just receive firebase from the arguments. In case that we change DBProvider in the future. Less dependencies